kafka/integration_test.go

```diff
@@ -0,0 +1,529 @@
+}
+
+var groupId = "integration-test-group.go"
+//var topic = "integration-test.go"
+
+//Test LibraryVersion()
+func TestLibraryVersion(t *testing.T) {
+	ver, verstr := LibraryVersion()
+	t.Logf("Library version %d: %s\n", ver, verstr)
+}
+
+//Test TimestampType
+func TestTimestampType(t *testing.T) {
+	timestamp_map := map[TimestampType]string{TimestampCreateTime: "CreateTime",
+		TimestampLogAppendTime: "LogAppendTime",
+		TimestampNotAvailable:  "NotAvailable"}
+	for ts, desc := range timestamp_map {
+		if ts.String() != desc {
+			t.Errorf("Wrong description")
+		}
+	}
+}
+
+//Test Offset APIs
+func TestOffsetAPIs(t *testing.T) {
+	offsets := []Offset{OffsetBeginning, OffsetEnd, OffsetInvalid, OffsetStored, 1001}
+	for _, offset := range offsets {
+		t.Logf("Offset: %s\n", offset.String())
+	}
+
+	// test known offset strings
+	testOffsets := map[string]Offset{"beginning": OffsetBeginning,
+		"earliest": OffsetBeginning,
+		"end":      OffsetEnd,
+		"latest":   OffsetEnd,
+		"unset":    OffsetInvalid,
+		"invalid":  OffsetInvalid,
+		"stored":   OffsetStored}
+	for key, expected_offset := range testOffsets {
+		offset, err := NewOffset(key)
+		if err != nil {
+			t.Errorf("Cannot create offset for %s\n", key)
+		} else {
+			if offset != expected_offset {
+				t.Errorf("Offset does not equal expected: %s != %s\n", offset, expected_offset)
+			}
+		}
+	}
+
+	// test numeric string conversion
+	offset, err := NewOffset("10")
+	if err != nil {
+		t.Errorf("Cannot create offset for 10\n")
+	} else {
+		if offset != Offset(10) {
+			t.Errorf("Offset does not equal expected: %s != %s\n", offset, Offset(10))
+		}
+	}
+
+	// test integer offset
+	var int_offset int = 10
+	offset, err = NewOffset(int_offset)
+	if err != nil {
+		t.Errorf("Cannot create offset for 10\n")
+	} else {
+		if offset != Offset(10) {
+			t.Errorf("Offset does not equal expected: %s != %s\n", offset, Offset(10))
+		}
+	}
+
+	// test int64 offset
+	var int64_offset int64 = 10
+	offset, err = NewOffset(int64_offset)
+	if err != nil {
+		t.Errorf("Cannot create offset for 10\n")
+	} else {
+		if offset != Offset(10) {
+			t.Errorf("Offset does not equal expected: %s != %s\n", offset, Offset(10))
+		}
+	}
+
+	// test invalid string offset
+	offset, err = NewOffset("what is this offset")
+	if err == nil {
+		t.Errorf("Expected error for this string offset\n")
+	}
+
+	// test double offset
+	offset, err = NewOffset(12.15)
+	if err == nil {
+		t.Errorf("Expected error for this double offset: 12.15\n")
+	}
+
+	// test change offset via Set()
+	offset, err = NewOffset("beginning")
+	if err != nil {
+		t.Errorf("Cannot create offset for 'beginning'\n")
+	}
+	err = offset.Set("latest")
+	if err != nil {
+		t.Errorf("Cannot set offset to 'latest'\n")
+	}
+
+	// test OffsetTail()
+	tail := OffsetTail(offset)
+	t.Logf("offset tail %v\n", tail)
+
+}
+
+//Test config map APIs
+func TestConfigMapAPIs(t *testing.T) {
+	if !testconfRead() {
```

Comment:
This test doesn't seem to need to connect to a real broker so the testconf shouldnt be used.